File: java-player/src/main/java/ubc/cosc322/messages/MCTSUpdateMessage.java
package ubc.cosc322.messages;

public class MCTSUpdateMessage extends BaseMessage {
    public MCTSUpdate[] updates;

    public MCTSUpdateMessage(MCTSUpdate[] updates) {
        message_type = MessageType.MCTS_UPDATE_MESSAGE;
        this.updates = updates;
    }
}

File: shared/messages/mcts-update-message.ts
import { MessageType } from "../enum/message-type";
import { MCTSUpdate } from "../models/gameplay/MCTSUpdate";
import { BaseMessage } from "./base-message";

export class MCTSUpdateMessage extends BaseMessage {
  public message_type = MessageType.MCTS_UPDATE_MESSAGE;
  public updates: MCTSUpdate[] = [];

  constructor(data: Partial<MCTSUpdateMessage>) {
    super();
    Object.assign(this, data);
  }
}
